### PR TITLE
WIP: Localization

### PR DESCRIPTION
**Pull request by EricRabil**
Here's the localization! I did my best to get all strings from the templates.

## Todo:

- [x] Server generated strings
- [ ] Client side strings
- [ ] API Responses?
- [ ] Other strings

### CODE REVIEW

**Comment by AppleBetas** [comment]
What does this do for text generated by client-side JS?

**Comment by EricRabil** [comment]
That's not taken into account in this, I can do that though (not tonight.) This PR only adds localization for EJS files.

**Comment by AppleBetas** [comment]
Alright, we should handle that before merging this.

File: .gitignore [hunk 1]
```diff
@@ -73,3 +73,5 @@ $RECYCLE.BIN/
 
 # Windows shortcuts
 *.lnk
+
+yarn.lock
```
Comment (nullpixel) [inline .gitignore]: yarn.lock should be committed

File: controllers/GuidelineController.js [hunk 2]
```diff
@@ -1,8 +1,18 @@
 const fs = require("fs");
 const marked = require("../util/Markdown");
+const path = require("path");
 
 exports.getGuidelines = (req, res, next) => {
-    fs.readFile(__dirname + "/../config/community_guidelines.md", "utf8", (err, data) => {
+    const guidelinesLocalePath = path.join(process.cwd(), "config", `community_guidelines_${res.locale}.md`);
```
Comment (nullpixel) [inline controllers/GuidelineController.js:6]: Can we move community guidelines out of the config folder if we're going to follow this pattern

File: locales/en.json [hunk 3]
```diff
@@ -0,0 +1,126 @@
+{
+	"language_name": "English",
```
Comment (nullpixel) [inline locales/en.json:2]: missing `translation_list_suffix`

File: yarn.lock [hunk 4]
```diff
@@ -1,4430 +0,0 @@
-# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (nullpixel) [inline yarn.lock]: yarn lockfiles should be committed https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/

**Review by nullpixel — CHANGES_REQUESTED** [review]
minor things. when merging this, please squash

File: locales/en.json (hunk 3 above)
Comment (AppleBetas) [reply locales/en.json:2]: My thought was that we only use it for crediting translators and otherwise the code should just display `language_name`

Comment (nullpixel) [reply locales/en.json:2]: Well, someone technically translated this

Comment (AppleBetas) [reply locales/en.json:2]: Us? lol

Comment (nullpixel) [reply locales/en.json:2]: yeah?

Comment (AppleBetas) [reply locales/en.json:2]: I mean we coded the rest of the site but I just thought it was more fitting so we could credit translators for their work (given that they're not part of Dynastic)

Comment (nullpixel) [reply locales/en.json:2]: Totally agree

**Comment by EricRabil** [comment]
This is ded

**Comment by nullpixel** [comment]
on hold for backend rewrite

**Pull request closed without merging**